test/broker_test/broker_test.go

```diff
@@ -1214,14 +1339,28 @@ var _ = test.DescribeTestsFor(test.TestCase{
 
 							assertInvocationCount(brokerServer.CatalogEndpointRequests, 1)
 						})
+
+						It("is returned from the repository in the brokers catalog field", func() {
+							ctx.SMWithOAuth.PATCH("/v1/service_brokers/" + brokerID).
+								WithJSON(common.Object{}).
+								Expect().
+								Status(http.StatusOK)
+
+							brokerFromDB, err := repository.Get(context.TODO(), types.ServiceBrokerType, brokerID)
+							Expect(err).ToNot(HaveOccurred())
+
+							Expect(string(brokerFromDB.(*types.ServiceBroker).Catalog)).To(MatchJSON(string(brokerServer.Catalog)))
+						})
 					})
 
 					Context("when an existing service plan is modified", func() {
 						Context("when catalog service plan id is modified but the catalog name is not", func() {
+							var expectedCatalog string
+
 							BeforeEach(func() {
-								sbCatalog := brokerServer.Catalog
+								expectedCatalog = string(brokerServer.Catalog)
 
-								catalog, err := sjson.Set(string(sbCatalog), "services.0.plans.0.id", "new-id")
+								catalog, err := sjson.Set(string(brokerServer.Catalog), "services.0.plans.0.id", "new-id")
```

Comment:
Why not use `expectedCatalog` here directly? (in this instance and in some places above)